### PR TITLE
feat(new device): adding support for Aqara Z1 Pro switches

### PR DESCRIPTION
**Pull request by mrskycriper**
This PR should close those issues:
- fix https://github.com/Koenkk/zigbee2mqtt/issues/21687
- fix https://github.com/Koenkk/zigbee2mqtt/issues/21668

Help with slider is strongly appreciated. Preferably with debug logs while touching the slider.

### CODE REVIEW

**Comment by Koenkk** [comment]
Looks good, let me know once this is ready for merge.

**Comment by mrskycriper** [comment]
> Looks good, let me know once this is ready for merge.

I'm mainly waiting for feedback on slider commands. Otherwise they are the same as Z1.

**Comment by mrskycriper** [comment]
> I would like to request for the following fix,
> 
>     * `slide_speed` should be in `mm/s` for units
> 
> 
> and an **optional** fix,
> 
>     * For `slide_direction`, `3` means that the user tapped on the slider without sliding. Recommend changing the label to 'tap' or 'none'.
> 
> 
> For **src/lib/lumi.ts**
> 
> ```js
> const exposes = [
>     e.numeric('slide_time', ea.STATE).withUnit('ms').withCategory('diagnostic'),
>     e.numeric('slide_speed', ea.STATE).withUnit('mm/s').withCategory('diagnostic'),
>     e.numeric('slide_relative_displacement', ea.STATE).withCategory('diagnostic'),
>     e.enum('slide_direction', ea.STATE, ['none', 'up', 'down']).withCategory('diagnostic'),
>     e.numeric('slide_time_delta', ea.STATE).withUnit('ms').withCategory('diagnostic'),
> ];
> ```

Sounds reasonable, I'll implement them.

**Comment by LZXCorp** [comment]
I was using the Light Switch slider for some automations, however one part of the automations didn't work properly.

Apparently, `652` (`slide_direction`) value when tapping can also be `1`
Give me a moment to test out all the possible values for `652`

**Comment by LZXCorp** [comment]
Seems like `652` not only states the action, it also states whether the person tapped, double tapped or hold onto the slider.
I'd like to recommend changing the entity name from `slide_direction` to `slide_action` given this new finding.

I'll update this message with all of the possible `652` values and states.

**Comment by mrskycriper** [comment]
> Seems like `652` not only states the tapping action, it also states whether the person double tapped or hold onto the slider. I'd like to recommend changing the entity name from `slide_direction` to `slide_action` given this new finding.
> 
> I'll update this message with all of the possible `652` values and states.

I see, that makes sense. What are the additional possible values then?

**Comment by mrskycriper** [comment]
How about naming it more generic `action`? With values being `single`, `double`, `hold`, `slide_up` and `slide_down`.

**Comment by LZXCorp** [comment]
> How about naming it more generic `action`? With values being `single`, `double`, `hold`, `slide_up` and `slide_down`.

Cause the individual light switches (i.e. the top and bottom switch) are already defined in an `action`

**Comment by mrskycriper** [comment]
> > How about naming it more generic `action`? With values being `single`, `double`, `hold`, `slide_up` and `slide_down`.
> 
> Cause the individual light switches (i.e. the top and bottom switch) are already defined in an `action` entity (example: `single_top`, `single_bottom`).

Oh, right, Aqara placed it on the same endpoint as the top one.

**Comment by LZXCorp** [comment]
Thanks for the help!

**Comment by mrskycriper** [comment]
I'll also add the pictures to the wiki soon

**Comment by LZXCorp** [comment]
I already added the images, I would also like to apologize, the action should also be null after the action is complete just like how the `action` entity works.

**Comment by mrskycriper** [comment]
> I already added the images.

That is very good. The pictures are high quality.


> I would also like to apologize, the `slide_action` should also be null after the action is complete just like how the `action` entity works. (No data will be sent when the action is done, just saying rn incase the `action` entity works like that)

Hmm, so it doesn't send anything as a filler? I'll have to see, how this works with action

**Comment by mrskycriper** [comment]
@Koenkk how to handle this type of situation? I don't see any obvious way to reset actions right after they are generated

**Comment by LZXCorp** [comment]
I'm currently checking on `Mode switch` and `LED disabled night` since those two seem like they aren't working as intended.

I'll report back within 24 hours with my findings.

**Comment by mrskycriper** [comment]
> I found out that there are sensitivity settings for the slider on Youtube, however, I do not know how to find the endpoints for it, any tips?

Unfortunately, I have no idea. It is probably on one of the numeric attributes of `manuSpecificLumi` on endpoint `1`.

**Comment by mrskycriper** [comment]
I've checked `led_disabled_night` and it uses `515` attribute. But it looks to be absent from the database entry provided in the issue. So it's either was moved or is missing entirely

**Comment by mrskycriper** [comment]
Attribute `617` is probably responsible for individual channel led behavior. But the values are unknown

**Comment by Koenkk** [comment]
Can't we just use `action` and then e.g. `slide_single`?

**Comment by mrskycriper** [comment]
> Can't we just use `action` and then e.g. `slide_single`?

That is possible, but how to merge it with other action provided by extend? It's possible values only include switch actions.

**Comment by MadMirza** [comment]
@mrskycriper @LZXCorp @Koenkk could you guys guide me on how I can implement the support for the slider? I have got the quadruple rocker version with me.
earlier I used an external converter for non pro Aqara Z1 found on one of the threads, everything except slider works with that converter. I have tried Z2M Edge just in case, but did not work.

I could help the way I can after adding the slider support to make it more perfect.

**Comment by MadMirza** [comment]
i do have a Aqara Hub, let me know if adding this to the hub can help in any sorts.

**Comment by mrskycriper** [comment]
> @mrskycriper @LZXCorp @Koenkk could you guys guide me on how I can implement the support for the slider? I have got the quadruple rocker version with me. earlier I used an external converter for non pro Aqara Z1 found on one of the threads, everything except slider works with that converter. I have tried Z2M Edge just in case, but did not work.
> 
> I could help the way I can after adding the slider support to make it more perfect.

Currently it's a bit of a mess. You can add it as an external converter, but I think we can't really do much except provide exposed raw values. Aqara calculates everything on the Hub from those raw values

**Comment by Koenkk** [comment]
> > Can't we just use `action` and then e.g. `slide_single`?
> 
> That is possible, but how to merge it with other action provided by extend? It's possible values only include switch actions.

We can add an additional parameter: `extraActions`

**Comment by mrskycriper** [comment]
> > > Can't we just use `action` and then e.g. `slide_single`?
> > 
> > 
> > That is possible, but how to merge it with other action provided by extend? It's possible values only include switch actions.
> 
> We can add an additional parameter: `extraActions`

That could work, I'll add this.

Also, what about action resets on other parameters?

**Comment by mrskycriper** [comment]
@LZXCorp is it enough to make it work if only action is resetting without other parameters?

**Comment by LZXCorp** [comment]
> i do have an Aqara Hub, let me know if adding this to the hub can help in any sorts.

You can help out with finding the endpoints for the slider sensitivity, and also help out for some of the other features I can't test cause I'm running the switch without a neutral wire.

> could you guys guide me on how I can implement the support for the slider?
I used the one in one of the older commits:

```
const lumiSlider = () => {
    const fromZigbee = [{
        cluster: 'manuSpecificLumi',
        type: ['attributeReport', 'readResponse'],
        convert: (model, msg) => {
            if (msg.data.hasOwnProperty(652)) {
                const actionLookup = {1: 'single', 2: 'double', 3: 'hold', 4: 'slide_up', 5: 'slide_down'};
                return {
                    slide_time: msg.data[561],
                    slide_speed: msg.data[562],
                    slide_relative_displacement: msg.data[563],
                    slide_action: actionLookup[msg.data[652]],
                    slide_time_delta: msg.data[769],
                };
            }
        },
    }];

    const exposes = [
        e.numeric('slide_time', ea.STATE).withUnit('ms').withCategory('diagnostic'),
        e.numeric('slide_speed', ea.STATE).withUnit('mm/s').withCategory('diagnostic'),
        e.numeric('slide_relative_displacement', ea.STATE).withCategory('diagnostic'),
        e.enum('slide_action', ea.STATE, ['single', 'double', 'hold', 'slide_up', 'slide_down']).withCategory('diagnostic'),
        e.numeric('slide_time_delta', ea.STATE).withUnit('ms').withCategory('diagnostic'),
    ];

    return {fromZigbee, exposes, isModernExtend: true};
}
```

Although I think they are now trying to combine `slide_action` into `action` so don't use it yet.

**Comment by LZXCorp** [comment]
> @LZXCorp is it enough to make it work if only action is resetting without other parameters?

I don't think it will be enough, I'll test the new stuffys first and report back.

**Comment by Koenkk** [comment]
Is this ready for merge?

**Comment by mrskycriper** [comment]
> Is this ready for merge?

Yes, I don't have anything else to add here and actions looks to be in working condition

File: src/lib/lumi.ts [hunk 1]
```diff
@@ -1589,6 +1609,107 @@ export const lumiModernExtend = {
...
+                if (msg.data.hasOwnProperty(652)) {
+                    const actionLookup: KeyValueNumberString = {
+                        1: 'slider_single',
+                        2: 'slider_double',
+                        3: 'slider_hold',
+                        4: 'slider_up',
+                        5: 'slider_down',
+                    };
+                    return {
+                        slide_time: msg.data[561],
```
Comment (Koenkk) [inline src/lib/lumi.ts:1660]: Can we prefix these all with `action_`? Then they will be automatically cleared.

Comment (mrskycriper) [reply src/lib/lumi.ts:1660]: So, that is how to do it. Got it, I'll do this now

**Comment by Koenkk** [comment]
Thanks!

**Comment by LZXCorp** [comment]
Oops, minding closing this again? Sorry.

**Comment by joshuakoh7** [comment]
Trying to add this device but getting errors
```
Info 2024-05-17 23:47:06Configuring '0x54ef441000bc473b'
Error 2024-05-17 23:47:16Failed to configure '0x54ef441000bc473b', attempt 5 (Error: Bind 0x54ef441000bc473b/2 genOnOff from '0x00124b001ca1bde7/1' failed (AREQ - ZDO - bindRsp after 10000ms) at Object.start (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:63:23) at /opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:691:28 at Queue.execute (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/queue.ts:35:20) at Endpoint.bind (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:464:13) at setupAttributes (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/lib/modernExtend.ts:75:13) at result.configure (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/lib/modernExtend.ts:417:17) at Object.configure (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/index.ts:172:21) at Configure.configure (/opt/zigbee2mqtt/lib/extension/configure.ts:121:13) at Configure.onMQTTMessage (/opt/zigbee2mqtt/lib/extension/configure.ts:55:21) at EventEmitter.wrappedCallback (/opt/zigbee2mqtt/lib/eventBus.ts:174:17))
```

**Comment by samuelthng** [comment]
Hello, pulling up this thread.
I just saw in the change logs for 1.38:
https://github.com/Koenkk/zigbee-herdsman-converters/pull/7536/files

Was wondering if we could use this as hints to port the same sensitivity feature over to the Z1 Pro's sliders.

**Pull request merged**